godot-core/src/builtin/string/macros.rs

```diff
@@ -0,0 +1,42 @@
+/*
+ * This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at https://mozilla.org/MPL/2.0/.
+ */
+
+#![macro_use]
+
+macro_rules! impl_rust_string_conv {
+    ($Ty:ty) => {
+        impl<S> From<S> for $Ty
+        where
+            S: AsRef<str>,
+        {
+            fn from(string: S) -> Self {
+                let intermediate = GodotString::from(string.as_ref());
+                Self::from(&intermediate)
+            }
+        }
```

Comment:
In the case of `$Ty == GodotString`, this would first call 
```rs
GodotString::from(&str)
``` 
and then 
```rs
GodotString::from(&GodotString)
```

Which trait impl powers the 2nd conversion? There is no `From<&GodotString> for GodotString`. While `From` is reflexive, it shouldn't be for references...?

The background behind my question is that it might perform an unnecessary conversion/copy.
Could you implement `from()` directly like this?
```rs
            fn from(string: S) -> Self {
                // note: no &
                Self::from(GodotString::from(string.as_ref()));
            }
```

Reply:
I don't use this macro for `GodotString`

Reply:
Ah, I see. In that case you can go ahead! 🙂